### ISSUE TITLE
Select multiple with search breaks

### ISSUE THREAD

**Issue by alansouzati** [grommet/grommet#2961]
### Expected Behavior
Select multiple with search should allow me to filter the list of options and keep selecting and unselecting things

### Actual Behavior
When using Select multiple and I perform a search, select breaks.

### URL, screen shot, or Codepen exhibiting the issue
https://codesandbox.io/s/2pop0013q0

### Steps to Reproduce

1. Open sandbox link
2. Open Select and choose option one
3. Open select again and this search search "two"
4. Select "two"
5. open select one last time.
6. Notice that only two is selected where one and two should be selected.

### Your Environment

- Grommet version: v2 latest
- Browser Name and version: any
- Operating System and version (desktop or mobile): any

**Comment by oorestisime** [comment]
There is a bigger issue here i think. The selected indexes are wrong when there is a search involved. It reflects the index relative to the search and not the absolute index.

**Comment by oorestisime** [comment]
And `options.indexOf(v)` for each value can return -1 if the value  is not in the search container.

**Comment by ShimiSun** [comment]
Yea, I noticed the same type of behavior and hence labeled the issue as `arch` (i.e architectural problem). 
whether you had previous values selected before the search, the values won't be available on the `options` array due to filtered options resulted from the search, hence you'll be losing their indexes. An idea I had was that we need a helper method that will be called after the search and before `selectOption` that will send the original values of indexes and options.

**Comment by oorestisime** [comment]
Another solution would be to store initial options in the state something like:

```
constructor(props) {
    super(props);
    this.state = {
      initialOptions: props.options,
    };
  }
```

and then calculate the selected with regards to that

**Comment by romanalesenkov1** [comment]
I can reproduce this issue now here: 
https://codesandbox.io/s/github/grommet/grommet-sandbox?initialpath=/select&module=/src/Select.js&file=/src/Select.js

**Comment by ShimiSun** [comment]
@romanalesenkov1 , thanks for reporting, reopening the issue...

**Comment by ShimiSun** [comment]
The fix is available @grommet version 2.16.0. Thank you @romanalesenkov1 for the fix.
